src/test/java/bdd/SpaceEngineersCucumberTest.kt

```diff
@@ -0,0 +1,73 @@
+package bdd
+
+import io.cucumber.java.After
+import io.cucumber.java.Before
+import io.cucumber.java.en.Given
+import io.cucumber.java.en.Then
+import io.cucumber.java.en.When
+import io.cucumber.junit.Cucumber
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.runner.RunWith
+import spaceEngineers.SeObservation
+import spaceEngineers.SeRequest
+import spaceEngineers.SpaceEngEnvironment
+import spaceEngineers.commands.ObservationArgs
+import spaceEngineers.commands.ObservationMode
+import spaceEngineers.commands.SeAgentCommand
+import testhelp.TEST_AGENT
+
+@RunWith(Cucumber::class)
+class SpaceEngineersCucumberTest {
+    lateinit var environment: SpaceEngEnvironment
+
+    val observations: MutableList<SeObservation> = mutableListOf()
+
+    @Before
+    fun setup() {
+        observations.clear()
+    }
+
+    @After
+    fun cleanup() {
+        observations.clear()
+        if (this::environment.isInitialized) {
+            environment.close()
+        }
+    }
+
+    @Given("I am connected to server.")
+    fun i_am_connected_to_mock_server() {
+        environment = SpaceEngEnvironment.localhost()
+    }
+
+    @When("I request for blocks.")
+    fun i_request_for_blocks() {
+        environment.getSeResponse(
+                SeRequest.command(SeAgentCommand.observe(TEST_AGENT, ObservationArgs(ObservationMode.BLOCKS)))
+        ).let { observations.add(it) }
+    }
+
+    @Then("I receive observation.")
+    fun i_receive_observation() {
+        assertTrue(observations.isNotEmpty())
+    }
+
+    @Then("I see {int} grid with {int} block.")
+    fun i_see_grid_and_with_block(grids: Int, blocks: Int) {
+        val observation = observations.last()
+        assertEquals(grids, observation.grids.size)
+        assertEquals(blocks, observation.grids[0].blocks.size)
+    }
+
+    @Then("Block with id {string} has {float} max integrity, {float} integrity and {float} build integrity.")
+    fun block_with_id_blk_has_max_integrity_integrity_and_build_integrity(
+            id: String, maxIntegrity: Float, integrity: Float, buildIntegrity: Float
+    ) {
+        val observation = observations.last()
+        val block = observation.grids.flatMap { it.blocks }.find { it.id == id } ?: error("block $id not found")
+        assertEquals(maxIntegrity, block.maxIntegrity)
+        assertEquals(buildIntegrity, block.buildIntegrity)
+        assertEquals(integrity, block.integrity)
```

Comment:
comparing floats on equality is generally not recommended

Reply:
Yeah, I see it as okay in testing since worst case scenario is, that some test fails while they should not. It's unlikely to create false positives.
Another option is to create assertEquals function with a delta parameter to check if those floats are close enough to eachother.